sdk/core/ts-http-runtime/src/client/clientHelpers.ts

```diff
@@ -69,7 +70,12 @@ export function createDefaultPipeline(
 ): Pipeline {
   const pipeline = createPipelineFromOptions(options);
 
-  pipeline.addPolicy(apiVersionPolicy(options));
+  if (options.apiVersion) {
```

Comment:
Hmm I wonder if we should just straight up remove this from the unbranded package right away? Otherwise we can make a note of this and do it when we're looking at the other breaking changes

Reply:
I think we probably should? There're some other things we might need to consider to remove or optimize from unbranded package 
like the deprecated baseUrl https://github.com/Azure/azure-sdk-for-js/blob/main/sdk/core/ts-http-runtime/src/client/common.ts#L318-L322
And the default logic for content type and request body serialization https://github.com/Azure/azure-sdk-for-js/issues/28487

Reply:
But maybe in the current stage, consistency between azure core and unbranded core has the higher priority?

Reply:
After rethinking, I feel like it might be problematic to add the warning here, because in the case that service does have a client level query apiVersion parameter, we will generate a client options with apiVersion in the generated code, 
```ts
export interface AnomalyDetectorRestClientOptions extends ClientOptions {
  apiVersion?: string;
}
```
in that case, customer can still pass { apiVersion: "" } to override the defined api version and it's expected, right?  

I kinda not sure how we can distinguish if the apiVersion is from ClientOptions in core and AnomalyDetectorRestClientOptions in the generated code ?

Reply:
there's another issue for removing this policy https://github.com/Azure/azure-sdk-for-js/issues/28204